### PR TITLE
Campaign signup endpoint

### PR DESCRIPTION
**Pull request by DFurnes**
Initial attempt at a `/campaigns/{id}/signup` endpoint. :construction: 

### Open Questions
 - How does Northstar authenticate to the Drupal API? Where do we handle permissions for things?
 - We're checking for an existing `Campaign` resource on Northstar, but some users may have signed up directly through Drupal. Is it bad that there are two error conditions for the same thing?
 - Is it unexpected/weird to have different behavior based on presence of the `user` parameter? I wonder if it makes more sense to have that always/never required.

/cc @angaither @jonuy

### CODE REVIEW

File: app/Northstar/Services/Drupal.php [hunk 1]
```diff
@@ -53,4 +52,28 @@ public function register($d_user)
...
+    $payload = [
+      'uid' => $drupal_id,
+      'source' => $source
+    ];
+
+    $response = $this->client->post('campaigns/' . $campaign_id . '/signup', [
+      'body' => json_encode($payload)
+    ]);
+
+    dd($response);
```
Comment (angaither) [inline app/Northstar/Services/Drupal.php]: pewwww

**Comment by angaither** [comment]
:cool:

File: app/Northstar/Services/Drupal.php (hunk 1 above)
Comment (DFurnes) [inline app/Northstar/Services/Drupal.php]: Whoopsie, but yeah totally not ready to merge. Just wanted to get the code out there so we could talk about it.

**Comment by jonuy** [comment]
> How does Northstar authenticate to the Drupal API? Where do we handle permissions for things?

As far as I know, we haven't created a way for Northstar to auth with Drupal. Two possible solutions:
- We do a login to get the session and token info and save those as ... environment variables? ... to include in all NS requests to Drupal.
- A more robust solution where Northstar actually does a login request to get that info, stores it in persistent storage, and logs in again or refreshes the token in the case that these things can expire.
- Or something in between.

> We're checking for an existing Campaign resource on Northstar, but some users may have signed up directly through Drupal. Is it bad that there are two error conditions for the same thing?

- Not sure I understand fully what you're talking about here...

> Is it unexpected/weird to have different behavior based on presence of the user parameter? I wonder if it makes more sense to have that always/never required.

- It's a bit of hidden logic that feels kinda iffy. I was wondering if it'd make sense to have different named endpoints. One that's meant for general submission of a signup or reportback, and one that's just meant for Drupal to update ids with.

**Comment by DFurnes** [comment]
Cleaned some things up, and going to merge this branch so it doesn't get too long-lived. The ` /campaigns/:id/signup` currently returns a `501 Not Implemented` error since it's... well, not implemented. :sweat_smile:

**Comment by angaither** [comment]
:+1:

**Pull request merged**